### PR TITLE
Origin update  dynamic user

### PR DESCRIPTION
**Pull request by rstijerina**
[v0.5-alpha](https://hub.docker.com/r/taccaci/docker-ubuntu-vnc-desktop-application-base/tags)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,154 +1,70 @@
-docker-ubuntu-vnc-desktop
+docker-ubuntu-vnc-desktop-application-base
 =========================
 
-[![Docker Pulls](https://img.shields.io/docker/pulls/dorowu/ubuntu-desktop-lxde-vnc.svg)](https://hub.docker.com/r/dorowu/ubuntu-desktop-lxde-vnc/)
-[![Docker Stars](https://img.shields.io/docker/stars/dorowu/ubuntu-desktop-lxde-vnc.svg)](https://hub.docker.com/r/dorowu/ubuntu-desktop-lxde-vnc/)
+Base docker image (HTML5 VNC interface to access Ubuntu 16.04 LXDE desktop environment) for applications. This image is used by interactive/gui applications.  Details on how to use this image to create an application can be found in [below](#use-base-image-to-create-an-application).
```
Comment (nathanfranklin) [inline README.md]:
```suggestion
Base docker image (HTML5 VNC interface to access Ubuntu 20.04 LXDE desktop environment) for applications. This image is used by interactive/gui applications.  Details on how to use this image to create an application can be found in [below](#use-base-image-to-create-an-application).
```

File: README.md [hunk 2]
```diff
@@ -1,154 +1,70 @@
...
+docker run -i --rm -p $port:6080 -e SSL_PORT=6080 -v $MYDATA:"/home/ubuntu/mydata" -e GUI_APPLICATION_DIRECTORY=/home/ubuntu/mydata  -e VNC_PASSWORD=1234 -e RESOLUTION="1080x720" --name "base_image_test_$AGAVE_JOB_OWNER"   -v /etc/pki/tls/certs/designsafe-exec-01.tacc.utexas.edu.cer:/etc/nginx/ssl/nginx.crt -v /etc/pki/tls/private/designsafe-exec-01.tacc.utexas.edu.key:/etc/nginx/ssl/nginx.key taccaci/docker-ubuntu-vnc-desktop-application-base:TAG
 ```
 
-Specify SSL port by `SSL_PORT`, certificate path to `/etc/nginx/ssl`, and forward it to 6081
+### Screen depth
 
-```
-docker run -p 6081:443 -e SSL_PORT=443 -v ${PWD}/ssl:/etc/nginx/ssl -v /dev/shm:/dev/shm dorowu/ubuntu-desktop-lxde-vnc
-```
+DISPLAY_SCREEN_DEPTH (i.e.`-e DISPLAY_SCREEN_DEPTH=24`) can be used to change the color depth which defaults to 16
```
Comment (nathanfranklin) [inline README.md]: This has been updated in their code so I think the default is now 24.  We had this in for STKO but now that the default is 24, hopefully we won't have to use this variable anymore (but good for now 👍 )

```suggestion
DISPLAY_SCREEN_DEPTH (i.e.`-e DISPLAY_SCREEN_DEPTH=16`) can be used to change the color depth which defaults to 24
```

File: README.md [hunk 3]
```diff
@@ -1,154 +1,70 @@
...
+mkdir temp_my_data
+docker run -p 6080:80 -e GUI_APPLICATION_DIRECTORY=/home/ubuntu/mydata -v temp_my_data:/home/ubuntu/mydata taccaci/docker-ubuntu-vnc-desktop-application-base:TAG
 ```
 
-Start the container
-
-```
-docker run -it --rm -p 6080:80 --device /dev/snd -e ALSADEV=hw:2,0 dorowu/ubuntu-desktop-lxde-vnc
+### User and group identity
+A user ubuntu is creaed with a home of `/home/ubuntu` a uid of `458981` and a gid of `816877`. The uid and gid can be set by the environment variable APP_USER_GROUP_ID and APP_USER_ID
```
Comment (nathanfranklin) [inline README.md]:
```suggestion
A user ubuntu is created with a home of `/home/ubuntu`. The uid (default: `-1001`) and gid (default: `-1000` can be set by the environment variable USER_GROUP_ID and USER_ID.
```

File: rootfs/startup.sh [hunk 4]
```diff
@@ -13,51 +13,78 @@ if [ -n "$X11VNC_ARGS" ]; then
...
+    if [ -z "$PASSWORD" ]; then
+        echo "  set default password to \"ubuntu\""
+        PASSWORD=ubuntu
+    fi
+    HOME=/home/$USER
+    echo "$USER:$PASSWORD" | chpasswd
+    cp -r /root/{.config,.gtkrc-2.0,.asoundrc} ${HOME}
+
+    # chown all files and directories excluding mounted dirs
+    find $HOME \( -path $HOME/mydata -prune -o \
```
Comment (nathanfranklin) [inline rootfs/startup.sh:41]: I don't have a suggestion but wondering about this list (community/projects/published/public).  I'm wondering what happens if we add something else in the future that we forget to add here?

File: README.md [hunk 5]
```diff
@@ -1,154 +1,70 @@
...
-3. Autostart, https://github.com/fcwu/docker-ubuntu-vnc-desktop/issues/85
-4. x11vnc arguments(multiptr), https://github.com/fcwu/docker-ubuntu-vnc-desktop/issues/101
-5. firefox/chrome crash (/dev/shm), https://github.com/fcwu/docker-ubuntu-vnc-desktop/issues/112
-6. resize display size without destroying container, https://github.com/fcwu/docker-ubuntu-vnc-desktop/issues/115#issuecomment-522426037
-
-License
-==================
-
-See the LICENSE file for details.
+docker run ... -e APP_USER_GROUP_ID=12345 -e APP_USER_ID=12345 ...
```
Comment (nathanfranklin) [inline README.md]:
```suggestion
docker run ... -e USER_GROUP_ID=12345 -e USER_ID=12345 ...
```

**Review by nathanfranklin — APPROVED** [review]
👍  💯 

I left some minor text changes.  Other than the question about the list of directories which we avoid calling chgrp on, this looks good to go! 🥇

File: rootfs/startup.sh (hunk 4 above)
Comment (rstijerina) [reply rootfs/startup.sh:41]: Totally. I was toying with some more dynamic solutions but decided to just roll with this hardcoded solution for now

**Pull request merged**